File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/foursquare/FourSquareSuite.kt
package cryptobench.ciphers.foursquare

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class FourSquareSuite : CipherSuite {
    override fun name(): String = "foursquare"

    override fun run(): SuiteResult {
        val cipher: Cipher = FourSquareCipher(FourSquareKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in FourSquareVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/foursquare/FourSquareSuite.scala
package cryptobench.ciphers.foursquare

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class FourSquareSuite extends CipherSuite {
  override def name: String = "foursquare"

  override def run(): SuiteResult = {
    val cipher: Cipher = new FourSquareCipher(FourSquareKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- FourSquareVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
